### PR TITLE
First version of camera ready version for liiv2021a

### PR DESCRIPTION
**Pull request by lennelov**
Hello, 
We get some errors when running 'make' (see https://pastebin.com/7anJ8MKR), but we seem to be able to get it to work by running 
```
make
biber article
latexmk -pdf -pdflatex="xelatex -interaction=nonstopmode" -use-make article.tex
```

We're not sure if this is the intended or an acceptable procedure, however.

### CODE REVIEW

File: liiv2021a/openreview/packages.tex [hunk 1]
```diff
@@ -0,0 +1,19 @@
+\usepackage{neurips_2019}
```
Comment (koustuvsinha) [inline liiv2021a/openreview/packages.tex:1]: Please remove the neurips_2019 package, as we are using ReScience styles. The compilation (with default make) fails due to this import.

Comment (lennelov) [reply liiv2021a/openreview/packages.tex]: Looks like we pushed the wrong version of our packages.tex, it should be fixed now.

File: liiv2021a/openreview/content.tex [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    \centering
+    \caption{Datasets used in the CIFAR-10 setting}
+    \addtolength{\leftskip} {-3cm}
+    \addtolength{\rightskip}{-3cm}
+    \begin{tabular}{r||c|c|c|c} 
+    \hline
+    Dataset & Samples & Classes & Dimensions & Link \\ [0.5ex] 
+    \hline
+    \hline
+    CIFAR-10 train & 50000 & 10 & 32x32x3 & \hrref{https://www.cs.toronto.edu/~kriz/cifar.html} \\
```
Comment (koustuvsinha) [inline liiv2021a/openreview/content.tex:92]: Compilation complains here (undefined control sequence). It should be `href` right?

File: liiv2021a/openreview/content.tex [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+\centering
+\caption{Training parameters in the CIFAR-10 setting}
+\addtolength{\leftskip} {-3cm}
+\addtolength{\rightskip}{-3cm}
+\begin{tabular}{l||c|c|c|c|c|c|c|c|c}
+\hline
+Model & Epochs & Cycle &  $\eta_0$ & $\eta_{max}$ & $\eta_{min}$ & Dropout & $T_0$ & Anneal & AUX data \\ [0.5ex] 
+\hline
+\hline
+DNN & 45 & 30 & 10^{-3} & 10^{-2} & 10^{-6} & 0.5 & - & - & - \\
```
Comment (koustuvsinha) [inline liiv2021a/openreview/content.tex:117]: Compilation issue: `! Missing $ inserted`. Please wrap the number `10^{-3}` in `$..$` -> `$10^{-3}`

File: liiv2021a/openreview/content.tex [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+EnD & 90 & 60 & 10^{-3} & 10^{-2} & 10^{-6} & 0.7 & 2.5 & No & - \\
```
Comment (koustuvsinha) [inline liiv2021a/openreview/content.tex:118]: Same here ^

File: liiv2021a/openreview/content.tex [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+EnD$_{\texttt{+AUX}}$ & 90 & 60 & 10^{-3} & 10^{-2} & 10^{-6} & 0.7 & 2.5 & No & CIFAR-100 \\
+$\text{EnD}^2$ & 90 & 60 & 10^{-3} & 10^{-2} & 10^{-6} & 0.7 & 10 & Yes & - \\
+EnD$^2_{\texttt{+AUX}}$ & 90 & 60 & 10^{-3} & 10^{-2} & 10^{-6} & 0.7 & 10 & Yes & CIFAR-100 \\
+PN & 45 & 30 & 0.5 \cdot 10^{-3} & 0.5 \cdot 10^{-2} & 0.5 \cdot 10^{-6} & 0.7 & - & No & CIFAR-100 \\
```
Comment (koustuvsinha) [inline liiv2021a/openreview/content.tex:122]: and in this whole table in fact ...

File: liiv2021a/openreview/supplementary.tex [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+    \caption{loss for EnD$^2$}
+    
+
+\end{algorithm} 
+\\
+
+
+
+
+\FloatBarrier
```
Comment (koustuvsinha) [inline liiv2021a/openreview/supplementary.tex:72]: Compilation complains this to be undefined

File: liiv2021a/openreview/supplementary.tex [hunk 7]
```diff
@@ -0,0 +1,532 @@
...
+    $\phi = En.predict(X)$ // exp($\phi$) are the labels for EnD$^2$  \\
+    // $\phi$ is a tensor of logits corresponding to the true distribution, each row corresponds to a model and each column a class. Each matrix corresponds to one image \\
+    $model_{\theta} \leftarrow classifier$ //create a new classifier model with weights $\theta$, with logits as output \\
+    EnD$^2 = argmin_{\theta}\{Loss_{EnD^2}(\phi,model_{\theta}(X)) \}$ //train model backpropagation\\
+    return EnD$^2$
+    \caption{Training algorithm for EnD$^2$ given an ensemble}
+    \label{alg:training}
+    
+\end{algorithm} \\
+\begin{algorithm}
```
Comment (koustuvsinha) [inline liiv2021a/openreview/supplementary.tex:37]: Please check the logs for compilation issues in this algorithm sections:

```
[11] (../openreview/supplementary.tex

Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 5.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `superscript' on input line 5.


Package hyperref Warning: Token not allowed in a PDF string (Unicode):
(hyperref)                removing `math shift' on input line 5.


! LaTeX Error: There's no line here to end.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.24     \Output{Trained EnD$^2$ model} \\
                                          

! LaTeX Error: There's no line here to end.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.37 \begin
           {algorithm}
! Double subscript.
<argument>  $\alpha _0_
                       i = \sum _j \alpha _{i,j}$ // sum over the classes to...
l.52     $
          P_{En} = softmax(\phi/T(t))$ // softmax over classes\\

! LaTeX Error: There's no line here to end.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.68 
     
! Undefined control sequence.
l.72 \FloatBarrier
                  
```

**Comment by koustuvsinha** [comment]
There are several compilation issues with the current changes. I would suggest you run `make` and try to isolate every error/warning shown in the logs. I pointed out the ones I noticed on my end. Please fix the above, and comment here if you are unsure about more errors, will help to isolate them further.

**Comment by lennelov** [comment]
> There are several compilation issues with the current changes. I would suggest you run `make` and try to isolate every error/warning shown in the logs. I pointed out the ones I noticed on my end. Please fix the above, and comment here if you are unsure about more errors, will help to isolate them further.

Thank you for identifying these errors! We think we have solved them now, and we can now run the make script locally without having to run biber manually.

**Comment by koustuvsinha** [comment]
Thanks for the fixes! I can confirm compilation works on my end now. Looks good to me! :) 

@jzf2101 can you review the paper contents? Thanks!

**Review by jzf2101 — APPROVED** [review]
The results in tables and plots look similar based on the PDF pre-built in the PR.  The authors provide an additional appendix not presented in the original submission, which while different I think helps contextualize the paper.

**Comment by toomasliiv** [comment]
> The results in tables and plots look similar based on the PDF pre-built in the PR. The authors provide an additional appendix not presented in the original submission, which while different I think helps contextualize the paper.

The additional appendix was present in the submission in the supplementary material, where the submission rules stated that it should be. See https://openreview.net/forum?id=p1BXNUcTFsN

**Pull request merged**